localstack-core/localstack/utils/event_matcher.py

```diff
@@ -0,0 +1,53 @@
+import json
+from typing import Any
+
+from localstack import config
+from localstack.services.events.event_ruler import matches_rule
+from localstack.services.events.v1.utils import matches_event as python_matches_event
+
+
+def matches_event(event_pattern: dict[str, Any] | str | None, event: dict[str, Any] | str) -> bool:
+    """
+    Match events based on configured rule engine.
+
+    Note: Different services handle patterns/events differently:
+    - EventBridge uses strings
+    - ESM and Pipes use dicts
+
+    Args:
+        event_pattern: Event pattern (str for EventBridge, dict for ESM/Pipes)
+        event: Event to match against pattern (str for EventBridge, dict for ESM/Pipes)
+
+    Returns:
+        bool: True if event matches pattern, False otherwise
+
+    Examples:
+        # EventBridge (string-based):
+        >>> pattern = '{"source": ["aws.ec2"]}'
+        >>> event = '{"source": "aws.ec2"}'
+
+        # ESM/Pipes (dict-based):
+        >>> pattern = {"source": ["aws.ec2"]}
+        >>> event = {"source": "aws.ec2"}
+
+    References:
+        - EventBridge Patterns: https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-patterns.html
+        - EventBridge Pipes: https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-pipes-event-filtering.html
+        - Event Source Mappings: https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html
+    """
+    if not event_pattern:
+        return True
+
+    if config.EVENT_RULE_ENGINE == "python":
```

Comment:
We noticed some bigger issues related to the Java-based implementation concerning the LocalStack shutdown behavior and decided it's better to switch the default to Python here. Hence, I would suggest to flip the conditional back to your original version.

I'm sorry for this back and forth 😢